### PR TITLE
Docs: Plugin E2E - Add authentication article

### PR DESCRIPTION
**Pull request by sunker**
**What this PR does / why we need it**:

This adds a new article that walks through authentication and RBAC. 

At the time of creating this docs PR, [this](https://github.com/grafana/plugin-tools/pull/726) PR is not yet merged, so you might get runtime errors if you try RBAC testing on your local machine. 

**Which issue(s) this PR fixes**:

Fixes #

Part of https://github.com/grafana/plugin-tools/issues/731

**Special notes for your reviewer**:

### CODE REVIEW

File: docusaurus/docs/test-a-plugin/guides/authentication.md [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+    dependencies: ['auth'],
+  },
+];
+```
+
+## Plugins that use RBAC
+
+If your plugin uses RBAC, you may want to write tests that verifies that certain plugin features are role-based. `@grafana/plugin-e2e` lets you define users with roles in the playwright config file. In the following example, a new user with the role `Viewer` is created in the `createViewerUserAndAuthenticate` setup project. In the next project, authentication state for the user with the viewer role is reused when running the tests. Note that tests that are specific for the `Viewer` role have been added to a dedicated `testDir`.
+
+```ts
```
Comment (leventebalogh) [inline docusaurus/docs/test-a-plugin/guides/authentication.md:60]: Maybe it worth adding [titles to the code-blocks](https://docusaurus.io/docs/markdown-features/code-blocks#code-title) to make it clearer which files they refer to.
```suggestion
```ts title="playwright.config.ts"
```

**Review by leventebalogh — APPROVED** [review]
Very nice one 👏

File: docusaurus/docs/test-a-plugin/guides/authentication.md (hunk 1 above)
Comment (sunker) [reply docusaurus/docs/test-a-plugin/guides/authentication.md]: Wasn't aware you could do that - thanks!

File: docusaurus/docs/e2e-test-a-plugin/authentication.md [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+sidebar_position: 40
+---
+
+# Introduction
+
+To be able to interact with the Grafana UI, you need to be logged in to Grafana. `@grafana/plugin-e2e` provides a declarative way to handle authentication and creating users that can be used to test role-based access control (RBAC) in your plugin.
+
+## Plugins that don't use RBAC
+
+If your plugin don't use RBAC, you can use the default server administrator credentials to login. The following snippet is a [setup project](https://playwright.dev/docs/test-global-setup-teardown#setup-example) that invokes a function in the `@grafana/plugin-e2e` package that will login to Grafana using `admin:admin`. The authenticated state is stored on disk and the file name pattern is as follows: `<plugin-root>/playwright/.auth/<username>.json`.
```
Comment (josmperez) [inline docusaurus/docs/e2e-test-a-plugin/authentication.md:22]:
```suggestion
If your plugin doesn't use RBAC, you can use the default server administrator credentials to login. The following snippet is a [setup project](https://playwright.dev/docs/test-global-setup-teardown#setup-example) that invokes a function in the `@grafana/plugin-e2e` package that will login to Grafana using `admin:admin`. The authenticated state is stored on disk and the file name pattern is as follows: `<plugin-root>/playwright/.auth/<username>.json`.
```

File: docusaurus/docs/e2e-test-a-plugin/authentication.md [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+      storageState: 'playwright/.auth/admin.json',
+    },
+    dependencies: ['auth'],
+  },
+];
+```
+
+## Plugins that use RBAC
+
+If your plugin uses RBAC, you may want to write tests that verifies that certain plugin features are role-based. `@grafana/plugin-e2e` lets you define users with roles in the playwright config file. In the following example, a new user with the role `Viewer` is created in the `createViewerUserAndAuthenticate` setup project. In the next project, authentication state for the user with the viewer role is reused when running the tests. Note that tests that are specific for the `Viewer` role have been added to a dedicated `testDir`.
```
Comment (josmperez) [inline docusaurus/docs/e2e-test-a-plugin/authentication.md:58]:
```suggestion
If your plugin uses RBAC, you may want to write tests that verify that certain plugin features are role-based. `@grafana/plugin-e2e` lets you define users with roles in the playwright config file. In the following example, a new user with the role `Viewer` is created in the `createViewerUserAndAuthenticate` setup project. In the next project, authentication state for the user with the viewer role is reused when running the tests. Note that tests that are specific for the `Viewer` role have been added to a dedicated `testDir`.
```

**Review by josmperez — APPROVED** [review]
LGTM.

**Review by sympatheticmoose — APPROVED** [review]
🎸

**Pull request merged**